DopplerCurrencyJob/DopplerCurrencyJob.cs

```diff
@@ -25,27 +29,85 @@ public class DopplerCurrencyJob
 
-        private async Task<IList<CurrencyResponse>> RunAsync()
+        private async Task<DopplerCurrencyJobResponse> RunAsync()
         {
-            _logger.LogInformation("Getting currency per each code enabled.");
+            var taskFailed = false;
+            var resultBuffer = new StringBuilder();
+            LogAndAppendToStringBuilder(resultBuffer, "Getting currency per each code enabled.");
+
             var currencyDto = await _dopplerCurrencyService.GetCurrencyByCode();
 
             if (!currencyDto.Any())
             {
-                _logger.LogInformation("Non-existent currencies for this date, please check for errors.");
-                return new List<CurrencyResponse>();
+                LogAndAppendToStringBuilder(resultBuffer, "Non-existent currencies for this date, please check for errors.");
+                return new DopplerCurrencyJobResponse()
+                {
+                    CurrencyList = new List<CurrencyResponse>(),
+                    Message = resultBuffer.ToString()
+                };
+            }
+
+            LogAndAppendToStringBuilder(resultBuffer, "Currency retrieved by code:");
+            AppendSerializedToStringBuilder(resultBuffer, currencyDto);
+
+            try
+            {
+                LogAndAppendToStringBuilder(resultBuffer, "Sending currency data to Doppler SAP system.");
+                var result = await _dopplerSapService.SendCurrency(currencyDto);
+
+                if (!result.IsSuccessStatusCode)
+                {
+                    LogAndAppendToStringBuilder(resultBuffer, $"{result.ReasonPhrase}. Error sending currency to Doppler SAP Api.");
+                    taskFailed = true;
+                }
+
+                LogAndAppendToStringBuilder(resultBuffer, "Sent currency data to Doppler SAP system.");
+            }
+            catch (Exception e)
+            {
+                LogAndAppendToStringBuilder(resultBuffer, $"An exception occurred when sending currency to Doppler SAP Api. Exception: {e.Message}");
+                taskFailed = true;
             }
 
-            _logger.LogInformation("Sending currency data to Doppler SAP system.");
-            await _dopplerSapService.SendCurrency(currencyDto);
+            try
+            {
+                LogAndAppendToStringBuilder(resultBuffer, "Insert currency data into Doppler Database.");
+
+                await _dopplerCurrencyService.InsertCurrencyIntoDataBase(currencyDto);
+
+                LogAndAppendToStringBuilder(resultBuffer, "Inserted currency data into Doppler Database.");
+            }
+            catch (Exception e)
+            {
+                LogAndAppendToStringBuilder(resultBuffer, $"An exception occurred when Insert currency data into Doppler Database. Exception: {e.Message}");
+                taskFailed = true;
+            }
+
+            if (taskFailed)
+            {
+                throw new DopplerCurrencyJobException(resultBuffer.ToString());
```

Comment:
I'm not sure it shows all the log.information in the public url, this information should be in the logging of the server

Reply:
I understand ..., but it is a way to identify that it is a manually thrown exception. In addition, it leaves open the possibility of personalizing it.

Reply:
Respondido en https://github.com/MakingSense/doppler-jobs/pull/84#discussion_r527958841-permalink